plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/ReserveVmInterfaceVspCommand.java

```diff
@@ -87,29 +114,285 @@ public boolean isDomainRouter() {
+            return this;
+        }
+
+        public Builder domainRouterIp(String domainRouterIp) {
+            this._domainRouterIp = domainRouterIp;
+            return this;
+        }
+
+        public Builder vmInstanceName(String vmInstanceName) {
+            this._vmInstanceName = vmInstanceName;
+            return this;
+        }
+
+        public Builder vmUuid(String vmUuid) {
+            this._vmUuid = vmUuid;
+            return this;
+        }
+
+        public Builder vmUserName(String vmUserName) {
+            this._vmUserName = vmUserName;
+            return this;
+        }
+
+        public Builder vmUserDomainName(String vmUserDomainName) {
+            this._vmUserDomainName = vmUserDomainName;
+            return this;
+        }
+
+        public Builder useStaticIp(boolean useStaticIp) {
+            this._useStaticIp = useStaticIp;
+            return this;
+        }
+
+        public Builder staticIp(String staticIp) {
+            this._staticIp = staticIp;
+            return this;
+        }
+
+        public Builder staticNatIpUuid(String staticNatIpUuid) {
+            this._staticNatIpUuid = staticNatIpUuid;
+            return this;
+        }
+
+        public Builder staticNatIpAddress(String staticNatIpAddress) {
+            this._staticNatIpAddress = staticNatIpAddress;
+            return this;
+        }
+
+        public Builder isStaticNatIpAllocated(boolean isStaticNatIpAllocated) {
+            this._isStaticNatIpAllocated = isStaticNatIpAllocated;
+            return this;
+        }
+
+        public Builder isOneToOneNat(boolean isOneToOneNat) {
+            this._isOneToOneNat = isOneToOneNat;
+            return this;
+        }
+
+        public Builder staticNatVlanUuid(String staticNatVlanUuid) {
+            this._staticNatVlanUuid = staticNatVlanUuid;
+            return this;
+        }
+
+        public Builder staticNatVlanGateway(String staticNatVlanGateway) {
+            this._staticNatVlanGateway = staticNatVlanGateway;
+            return this;
+        }
+
+        public Builder staticNatVlanNetmask(String staticNatVlanNetmask) {
+            this._staticNatVlanNetmask = staticNatVlanNetmask;
+            return this;
+        }
+
+        @Override
+        public ReserveVmInterfaceVspCommand build() {
+            return new ReserveVmInterfaceVspCommand(_nicUuid, _nicMacAddress, _networkUuid, _isL3Network, _isSharedNetwork, _vpcUuid, _networkDomainUuid, _networksAccountUuid,
+                    _isDomainRouter, _domainRouterIp, _vmInstanceName, _vmUuid, _vmUserName, _vmUserDomainName, _useStaticIp, _staticIp, _staticNatIpUuid, _staticNatIpAddress,
+                    _isStaticNatIpAllocated, _isOneToOneNat, _staticNatVlanUuid, _staticNatVlanGateway, _staticNatVlanNetmask);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
This ``equals`` implementation should include attributes from the ``Command`` super class.